src/js_realm.hpp

```diff
@@ -563,6 +563,10 @@ void RealmClass<T>::delete_one(ContextType ctx, ObjectType this_object, size_t a
 
     if (Object::template is_instance<RealmObjectClass<T>>(ctx, arg)) {
         auto object = get_internal<T, RealmObjectClass<T>>(arg);
+        if (!object->is_valid()) {
+            throw std::runtime_error("Object is invalid. Was it already deleted?");
```

Comment:
Maybe change message to:
"Object is invalid. Either it has been previously deleted or the Realm it belongs to has been closed."